### PR TITLE
fix: resolve owner_id compliance

### PR DESCRIPTION
**Pull request by evergreen-trading-systems**
https://github.com/Mintbase/omni-site/issues/1084

fixes https://github.com/Mintbase/mintbase-core/issues/8

### CODE REVIEW

**Comment by tifrel** [comment]
Why add this internal/external repetition and complexity? We should make sure that `enum Owner` serializes properly.

**Review by tifrel — CHANGES_REQUESTED** [review]
Why add this internal/external repetition and complexity? We should make sure that enum Owner serializes properly.

**Comment by evergreen-trading-systems** [comment]
> Why add this internal/external repetition and complexity? We should make sure that enum Owner serializes properly.

Because you cannot serialise `Owner` to string.

File: mintbase-deps/src/impls.rs [hunk 1]
```diff
@@ -1732,8 +1749,8 @@ impl MintbaseStore {
     pub fn nft_token(
         &self,
         token_id: U64,
-    ) -> Option<TokenCompliant> {
-        Some(self.nft_token_compliant_internal(token_id.0))
+    ) -> Option<TokenCompliantExternal> {
+        Some(self.nft_token_compliant_internal(token_id.0).to_external())
```
Comment (evergreen-trading-systems) [inline mintbase-deps/src/impls.rs:1753]: re: https://github.com/Mintbase/mintbase-core/pull/11/files#pullrequestreview-884908216

Without `to_external` it is impossible to serialise `Owner` as String. You either need to refactor all contracts or create new struct.

**Comment by evergreen-trading-systems** [comment]
> `impl Serialize for Owner { ... }` is neither a new struct, nor does it refactor all the contracts. We already had minor problems with contract size on last upgrade, and introducing another struct will further increase size. If you're busy with things like pipeline, I can take care of this.

As explained elsewhere. Code duplication is not an argument to a codebase that already has plenty. Therefore this a subjective matter on how much code duplication should be allowed. Such discussion is unrelated to this pr.

**Comment by tifrel** [comment]
Yup. But it's not subjective that duplication should be minimized. Some goes for storage requirements. As you mentioned other explanations elsewhere: This review regards this PR, not the rest of the codebase.

**Comment by evergreen-trading-systems** [comment]
> Yup. But it's not subjective that duplication should be minimized. Some goes for storage requirements. As you mentioned other explanations elsewhere: This review regards this PR, not the rest of the codebase.

I think you need to review your comments or what subjective means. What should be minimised or allowed is a discussion outside the scope of this pr. You will find in tools if you've used them like Sonarqube, they measure subjective issue like code duplication and other code smells against an acceptable level. There is no such tooling used here so please save everyone time arguing over issues like this.

**Comment by evergreen-trading-systems** [comment]
If the pr passes the pipeline, does what it says and does not jeopardise code integrity - I think that sounds like a candidate to merge. We can't effectively talk about subjective issues without bringing in to question many more topics putting such discussion out of scope for this pr.

**Pull request closed without merging**